medical/models/medical_patient.py

```diff
@@ -1,105 +1,124 @@
+from openerp.exceptions import ValidationError
 from datetime import datetime
-from openerp.tools.translate import _
 from dateutil.relativedelta import relativedelta
 
 
 class MedicalPatient(models.Model):
-    '''
-    The concept of Patient included in medical.
-    '''
     _name = 'medical.patient'
     _description = 'Medical Patient'
-    _inherits = {'res.partner': 'partner_id', }
-
-    @api.one
-    def _compute_age(self):
-        """
-        age computed depending of the birth date of the
-        membership request
-        """
-        now = datetime.now()
-        if self.dob:
-            dob = fields.Datetime.from_string(self.dob)
-
-            if self.deceased:
-                dod = fields.Datetime.from_string(self.dod)
-                delta = relativedelta(dod, dob)
-                deceased = _(' (deceased)')
-            else:
-                delta = relativedelta(now, dob)
-                deceased = ''
-            years_months_days = str(delta.years) + _('y ') + str(
-                delta.months) + _('m ') + str(delta.days) + _('d')\
-                + deceased
-        else:
-            years_months_days = _('No DoB !')
-        self.age = years_months_days
-
-    @api.one
-    def action_invalidate(self):
-        self.active = False
-        self.partner_id.active = False
+    _inherits = {'res.partner': 'partner_id'}
 
-    age = fields.Char(compute='_compute_age')
+    age = fields.Char(
+        compute='_compute_age',
+    )
     identification_code = fields.Char(
         string='Internal Identification',
-        help='Patient Identifier provided by the Health Center.'
-        '(different from the Social Security Number)')
-    general_info = fields.Text(string='General Information')
-    dob = fields.Date(string='Date of Birth')
-    dod = fields.Datetime(string='Deceased Date')
-    active = fields.Boolean(default=True)
-    deceased = fields.Boolean()
+        help='Patient identifier provided by the health center'
+             '(Different from the social security number)',
+    )
+    general_info = fields.Text(
+        string='General Information',
+    )
+    dob = fields.Date(
+        string='Date of Birth',
+    )
+    dod = fields.Datetime(
+        string='Deceased Date',
+    )
+    active = fields.Boolean(
+        default=True,
+    )
+    deceased = fields.Boolean(
+        compute='_compute_deceased',
+        store=True,
+        help='Automatically True if deceased date is set',
+    )
     partner_id = fields.Many2one(
-        comodel_name='res.partner', required=True, ondelete='cascade',
-        index=True)
-    gender = fields.Selection(
-        selection=[
-            ('m', 'Male'),
-            ('f', 'Female'),
-        ])
+        string='Related Partner',
+        comodel_name='res.partner',
+        required=True,
+        ondelete='cascade',
+        index=True,
+    )
+    gender = fields.Selection([
+        ('m', 'Male'),
+        ('f', 'Female'),
+    ], )
```

Comment:
Putting the `)` on the same line as the `]` defeats the purpose of the trailing `,`, which is to allow for addition of attributes without changing unrelated lines of code. Should be:
```
fields.Selection([
        ...
    ],
)
```